### PR TITLE
initial module commit

### PR DESCRIPTION
**Pull request by maximmi**
## what
* initial implementation of module

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,13 +1,65 @@
...
+  encrypted                           = var.encrypted
+  enhanced_vpc_routing                = var.enhanced_vpc_routing
+  kms_key_id                          = var.kms_key_arn
+  elastic_ip                          = var.elastic_ip
+  skip_final_snapshot                 = var.skip_final_snapshot
+  final_snapshot_identifier           = var.final_snapshot_identifier
+  snapshot_identifier                 = var.snapshot_identifier
+  snapshot_cluster_identifier         = var.snapshot_cluster_identifier
+  owner_account                       = var.owner_account
+  iam_roles                           = []
```
Comment (jamengual) [inline main.tf:30]: can we add an input variable to allow to pass the iam_roles?

File: variables.tf [hunk 2]
```diff
@@ -1,4 +1,182 @@
...
+variable "cluster_identifier" {
+  type        = string
+  default     = ""
+  description = "The Redshift Cluster Identifier. Must be a lower case string. Will use generated label ID if not supplied"
 }
+
+variable "database_name" {
+  type        = string
+  default     = "dev"
+  description = "The name of the first database to be created when the cluster is created."
```
Comment (jamengual) [inline variables.tf:10]: should we add `.` at the end of all the descriptions?

File: variables.tf [hunk 3]
```diff
@@ -1,4 +1,182 @@
...
+variable "snapshot_identifier" {
+  type        = string
+  default     = null
+  description = "The name of the snapshot from which to create the new cluster."
+}
+
+variable "snapshot_cluster_identifier" {
+  type        = string
+  default     = null
+  description = "The name of the cluster the source snapshot was created from."
```
Comment (jamengual) [inline variables.tf:144]: I'm a bit confused about the name of this variable and the description.
is this the snapshot to create a new cluster from ?

Comment (osterman) [reply variables.tf:144]: I think it's based on the upstream documentation https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/redshift_cluster#snapshot_cluster_identifier

**Review by osterman — APPROVED** [review]

**Pull request merged**